lib/elixir/lib/exception.ex

```diff
@@ -1342,8 +1342,39 @@ defmodule UndefinedFunctionError do
       hint_for_loaded_module(module, function, arity, nil)
   end
 
-  defp hint(_module, _function, _arity, _loaded?) do
-    ""
+  defp hint(module, function, arity, _loaded?) do
+    normalized_module = normalized_module_name(module)
+
+    candidate =
+      :code.all_available()
+      |> Enum.find(fn {name, _, _} ->
+        normalized_module_name(name) == normalized_module
+      end)
+
+    with {_, _, _} <- candidate,
+         {:module, module} <- load_module_filename(candidate),
+         true <- function_exported?(module, function, arity) do
+      ". Did you mean:\n\n      * #{Exception.format_mfa(module, function, arity)}\n"
+    else
+      _ -> ""
+    end
+  end
+
+  defp load_module_filename({name, path, loaded?}) do
```

Comment:
@josevalim please let me know if there's a better way to handle this. I read through all the functions exposed in [:code](https://www.erlang.org/doc/man/code#all_available-0) and [Code](https://hexdocs.pm/elixir/Code.html#ensure_loaded/1), but there doesn't seem to be a better way that I'm seeing

Reply:
```
      path
      |> Path.rootname(".beam")
      |> String.to_atom()
      |> Code.ensure_loaded()
```

:)

Reply:
I was worried about doing `to_atom`, but since its already coming from the compiler I suppose there's not really a concern. (and btw you're mixing the module name and path flows, but I know what you mean)

Reply:
@josevalim updated